waspc/src/Wasp/Version.hs

```diff
@@ -5,6 +5,6 @@ import qualified Paths_waspc
 import qualified Wasp.SemanticVersion as SV
 
 waspVersion :: SV.Version
-waspVersion = SV.Version (toEnum major) (toEnum minor) (toEnum patch)
-  where
-    DV.Version [major, minor, patch] _ = Paths_waspc.version
+waspVersion = case Paths_waspc.version of
+  DV.Version [major, minor, patch] _ -> SV.Version (toEnum major) (toEnum minor) (toEnum patch)
+  _ -> error "This should never happen. Wasp binary version have exactly three digits, but it doesn't."
```

Comment:
Another non-exhaustive pattern warning.

Btw, this got me wondering: What happens if we mistakenly forget to specify the minor or patch version number in `waspc.cabal`. When would it fail?

And the answer is: right here.
<img width="1686" height="640" alt="image" src="https://github.com/user-attachments/assets/05deb6ee-a316-457b-9a9c-4433083762f2" />

Without this new message, the error is a generic "non-exaustive pattern in..."

Reply:
I don't know, but I wouldn't worry too much, doesn't sound like a potential issue, right? If we do, it will fail in development anyway.

Reply:
You say "I don't know" but I didn't ask a question :)

I wouldn't know what else to do anyway.